<p>On a Thread instance, methods <code>wait(...)</code>, <code>notify()</code> and <code>notifyAll()</code> are available only because all classes in Java extends <code>Object</code> and so automatically inherit from those methods. There are two very good reasons to not call those methods on a Thread instance :</p>
<ul>
  <li>This is really confusing because what is really expected for instance when calling the <code>wait(...)</code> method on a Thread instance ? To suspend the execution of the Thread or to really wait for the acquisition of the monitor on this object ?</li>
  <li>Internally, the JVM relies on those method calls to change the state of the Thread (<code>BLOCKED</code>, <code>WAITING</code>, ...), so calling those methods will corrupt the behavior of the JVM</li>
  </ul>

<h2>Noncompliant Code Example</h2>

<pre>
Thread myThread = new Thread(new RunnableJob());
...
myThread.wait(2000);
</pre>
