<p>When a class implements an <code>Interface</code>, all of its subclasses automatically inherit that implementation, so there's no need to explicitly re-implement a superclass' <code>Interface</code>s in a child class. Simply <code>@Override</code> the parent class' <code>Interface</code> methods.</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class Fruit implements Fiberous {
  @Override
  public int getFiberCount() { ... }
}

public class Raspberry extends Fruit implements Fiberous { // Noncompliant
  @Override
  public int getFiberCount() { ... }
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public class Fruit implements Fiberous {
  @Override
  public int getFiberCount() { ... }
}

public class Raspberry extends Fruit {
  @Override
  public int getFiberCount() { ... }
}
</pre>

