<p>
If a private field is declared but not used in the program, it can be considered dead code and should therefore be removed.
This will improve maintainability because developers will not wonder what the variable is used for.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass {
  private int foo = 42;                       // foo is unused

  public int compute(int a) {
    return a * 42;
  }
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public class MyClass {

  public int compute(int a) {
    return a * 42;
  }
}
</pre>
