<p>Since arrays do not override Object.equals(), calling equals on two arrays is the same as comparing their addresses. This means that array1.equals(array2) is equivalent to array1==array2.</p>

<p>However, some developers might expect Array.equals(Object obj) to do more than a simple memory address comparison, comparing for instance the size and content of the two arrays. To prevent such a misunderstanding, the '==' operator must always be used in place of the Array.equals(Object obj) method.</p>

<h2>Noncompliant Code Example</h2>
<pre>
String[] array1 = ...
String[] array2 = ...
...
if(array1.equals(array2)){
...
}
</pre>

<h2>Compliant</h2>
<pre>
String[] array1 = ...
String[] array2 = ...
...
if(array1 == array2){
...
}
</pre>