<p>There are two ways to write a lambda that simply returns the result of an expression, but one is definitely more compact and readable than the other.</p>

<h2>Noncompliant Code Example</h2>

<pre>
 x -> {System.out.println(x+1);}
(a, b) -> { return a + b; }
</pre>

<h2>Compliant Solution</h2>

<pre>
 x -> {System.out.println(x+1);}
(a, b) -> a + b
</pre>
